nmostesting/mocks/Registry.py

```diff
@@ -271,3 +404,184 @@ def heartbeat(version, node_id):
+    if NMOSUtils.compare_api_version("v1.3", version) >= 0:
+        link = ""
+        if new_until != MAX_UNTIL:
+            link += "<http://localhost:5102/x-nmos/query/v1.3/" + resource_type + "s/?paging.since=" + new_until + "&paging.limit=" + str(registry.paging_limit) + ">; rel=\"next\""
+        
+        if since != MIN_SINCE:
+            if link != "":
+                link += ","
+            link += "<http://localhost:5102/x-nmos/query/v1.3/" + resource_type + "s/?paging.since=0:0&paging.limit=" + str(registry.paging_limit) + ">; rel=\"first\""
+
+        if link != "": 
+            response.headers["Link"] = link
+        response.headers["X-Paging-Limit"] = registry.paging_limit 
+        response.headers["X-Paging-Since"] = since
+        response.headers["X-Paging-Until"] = new_until
+        response.headers["X-Ancestry-Generations"] = 1
+
+    return response
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>/<resource>/<resource_id>', methods=['GET'], strict_slashes=False)
+def get_resource(version, resource, resource_id):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    registry.query_api_called = True
+
+    resource_type = resource.rstrip("s")
+    data = []
+    try:
+        # Type may not be in the list, so this could throw an exception
+        data = registry.get_resources()[resource_type][resource_id]
+    except Exception:
+        pass
+
+    return Response(json.dumps(data), mimetype='application/json')
+
+@REGISTRY_API.route('/x-nmos/query/<version>/subscriptions', methods=["POST"])
+def post_subscription(version):
+    
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    subscription_request = request.json
+
+    subscription_response = {}
+    created = False
+    
+    try:
+        subscription, created = registry.subscribe_to_query_api(version, subscription_request["resource_path"])
+
+        # Note: 'secure' not required in request, but is required in response
+        secure = subscription_request['secure'] if 'secure' in subscription_request else False
+
+        subscription_response = {'id': subscription["id"],
+            'max_update_rate_ms': subscription_request['max_update_rate_ms'],
+            'params': subscription_request['params'],
+            'persist': subscription_request['persist'],
+            'resource_path': subscription['resource_path'],
+            'secure': secure, 
+            'ws_href': subscription['ws_href'] }
+
+    except SubscriptionException as e:
+        print('Subscription failed: ' + e.args[0])
+
+    status_code = 200 if created else 201
```

Comment:
Status codes the other way around?

Reply:
Looks like it to me.

(Potentially ought to be picked up by the IS-04-02 test suite run against the mock?)

Reply:
Now fixed